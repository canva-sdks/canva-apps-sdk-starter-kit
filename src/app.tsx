import { Button, Rows, Text } from "@canva/app-ui-kit";
import { FormattedMessage, useIntl } from "react-intl";
import * as styles from "styles/components.css";
import { useAddElement } from "utils/use_add_element";

export const App = () => {
  const addElement = useAddElement();
  const onClick = () => {
    addElement({
      type: "text",
      children: ["Hello world!"],
    });
  };

  const intl = useIntl();

  return (
    <div className={styles.scrollContainer}>
      <Rows spacing="2u">
        <Text>
          <FormattedMessage
            defaultMessage="
              To make changes to this app, edit the <code>src/app.tsx</code> file,
              then close and reopen the app in the editor to preview the changes.
            "
            description="Instructions for how to make changes to the app. Do not translate <code>src/app.tsx</code>."
            values={{
              code: (chunks) => <code>{chunks}</code>,
            }}
          />
        </Text>
        <Button variant="primary" onClick={onClick} stretch>
          {intl.formatMessage({
            defaultMessage: "Do something cool",
            description:
              "Button text to do something cool. Creates a new text element when pressed.",
          })}
        </Button>
      </Rows>
    </div>
  );
};
