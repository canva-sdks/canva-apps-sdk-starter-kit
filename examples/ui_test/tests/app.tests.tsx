import { fireEvent, render } from "@testing-library/react";
import { TestAppUiProvider } from "@canva/app-ui-kit";
import { App } from "../app";

describe("app", () => {
  let addElementAtPoint: jest.Mock;

  beforeEach(() => {
    addElementAtPoint = jest.fn();
  });

  it("calls addElementAtPoint onClick", async () => {
    const result = render(
      // In a test environment, you should wrap your apps in `TestAppUiProvider`, rather than `AppUiProvider`
      <TestAppUiProvider theme="dark">
        <App onClick={addElementAtPoint} />
      </TestAppUiProvider>,
    );

    // get a reference to the button element
    const button = result.getByRole("button");

    // assert its label matches what we expect
    expect(button.textContent).toEqual("Do something cool");
    // assert our callback has not yet been called
    expect(addElementAtPoint).not.toHaveBeenCalled();

    // programmatically simulate clicking the button
    fireEvent.click(button);

    // assert our callback was called
    expect(addElementAtPoint).toHaveBeenCalledTimes(1);
  });

  it("Renders ðŸŽ‰", () => {
    const result = render(
      // In a test environment, you should wrap your apps in `TestAppUiProvider`, rather than `AppUiProvider`
      <TestAppUiProvider theme="dark">
        <App onClick={addElementAtPoint} />
      </TestAppUiProvider>,
    );

    expect(result.container).toMatchSnapshot();
  });
});
