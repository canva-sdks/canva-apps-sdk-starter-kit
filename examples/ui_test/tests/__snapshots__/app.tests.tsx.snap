// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app Renders ðŸŽ‰ 1`] = `
<div>
  <div
    class="scrollContainer"
  >
    <div
      class="jv_R6g"
      style="--NZu1Zw: 16px;"
    >
      <p
        class="D6xAvw E44aNA rU8RQQ"
      >
        This example demonstrates how to test your App's UI.
      </p>
      <p
        class="D6xAvw E44aNA rU8RQQ"
      >
        Checkout 
        <code>
          examples/ui_test/tests/app.tests.tsx
        </code>
         to learn how to start testing.
      </p>
      <button
        class="_5KtATA LQzFZw xwTbNA _8ERLTg rV61VQ LQzFZw xwTbNA _4_iekA j34Zww aqbYPg _3FvZZg"
        type="button"
      >
        <span
          class="khPe7Q"
        >
          Do something cool
        </span>
      </button>
    </div>
  </div>
</div>
`;
